crates/alpen-ee/sequencer/src/ol_chain_tracker/task.rs

```diff
@@ -0,0 +1,185 @@
+
+pub(crate) enum OlChainTrackerQuery {
+    GetFinalizedBlock(oneshot::Sender<OLBlockCommitment>),
+    GetInboxMessages {
+        from_slot: u64,
+        to_slot: u64,
+        response_tx: oneshot::Sender<eyre::Result<Vec<MessageEntry>>>,
+    },
+}
+
+pub(crate) async fn ol_chain_tracker_task<
+    TClient: SequencerOlClient,
+    TStorage: ExecBlockStorage,
+>(
+    mut chainstatus_rx: watch::Receiver<OLFinalizedStatus>,
+    mut query_rx: mpsc::Receiver<OlChainTrackerQuery>,
+    mut state: OlChainTrackerState,
+    client: Arc<TClient>,
+    storage: Arc<TStorage>,
+) {
+    loop {
+        select! {
+            chainstatus_changed = chainstatus_rx.changed() => {
+                if chainstatus_changed.is_err() {
+                    warn!("channel is closed; shutting down");
+                    break;
+                }
+                // we only track inbox messages from finalized blocks to include in block assembly.
+                let ol_status = *chainstatus_rx.borrow_and_update();
+                handle_chain_update(ol_status, &mut state, client.as_ref(), storage.as_ref()).await;
+            }
+            maybe_query = query_rx.recv() => {
+                let Some(query) = maybe_query else {
+                    warn!("channel is closed; shutting down");
+                    break;
+                };
+                handle_chain_query(&state, query);
+            }
+        }
+    }
+}
+
+fn handle_chain_query(state: &OlChainTrackerState, query: OlChainTrackerQuery) {
+    match query {
+        OlChainTrackerQuery::GetFinalizedBlock(tx) => {
+            let _ = tx.send(state.best_block());
+        }
+        OlChainTrackerQuery::GetInboxMessages {
+            from_slot,
+            to_slot,
+            response_tx,
+        } => {
+            let _ = response_tx.send(state.get_inbox_messages(from_slot, to_slot));
+        }
+    }
+}
+
+async fn handle_chain_update(
+    ol_status: OLFinalizedStatus,
+    state: &mut OlChainTrackerState,
+    client: &impl SequencerOlClient,
+    storage: &impl ExecBlockStorage,
+) {
+    // compare latest finalized block with local chain segment using db. get extend, revert info
+    match track_ol_state(state, ol_status, client).await {
+        Ok(TrackAction::Extend(ol_blocks)) => {
+            // update state
+            for OLBlockData {
+                commitment,
+                inbox_messages,
+            } in ol_blocks
+            {
+                state.append_block(commitment, inbox_messages).unwrap();
+            }
+
+            if let Err(err) = handle_state_pruning(state, ol_status, storage).await {
+                error!(?err, "failed to prune state");
+            }
+        }
+        Ok(TrackAction::Reorg(_next)) => {
+            // kill task and trigger app shutdown through TaskManager.
+            panic!("Deep reorg detected. Manual resolution required.")
+        }
+        Err(err) => {
+            error!("failed to track ol state; {}", err);
```

Comment:
Like the one above and following the logging best practices standards:
```suggestion
           error!(?err, "failed to track OL state");
```